app/src/main/java/de/rampro/activitydiary/ui/history/HistoryActivity.java

```diff
@@ -403,4 +441,60 @@ public void addDetailAdapter(long diaryEntryId, DetailRecyclerViewAdapter adapte
 
     }
 
+    /** Checks date format and also checks date can be parsed (used for not existing dates like 35.13.2000)
+     * @param date input that is checked
+     * @param dateFormat format under we want to check (parse) entered date
+     */
+    private boolean checkDateFormat(String date, String dateFormat){
+        if (dateFormat == null)
+            return false;
+
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dateFormat);
+        simpleDateFormat.setLenient(false);
+        switch (dateFormat) {
+            case "yyyy.MM.dd":
+                if (!date.matches("([0-9]{4})\\.([0-9]{1,2})\\.([0-9]{1,2})"))
+                    return false;
+                break;
+            case "dd.MM.yyyy":
+                if (!date.matches("([0-9]{1,2})\\.([0-9]{1,2})\\.([0-9]{4})"))
+                    return false;
+                break;
+            default:
+                Assert.fail("Wrong date format !");
+                return false;
+        }
+        try {
```

Comment:
this part here looks better, but isn't it dead code (I.e. never reached due to all the returns above in the swtich case?)